### ISSUE TITLE
Unable to send batch after restarting promtail on one of nodes

### ISSUE THREAD

**Issue by kovetskiy** [grafana/loki#167]
Today I did some maintenance with kubernetes node and I stopped promtail (actually just deleted the pod) on this instance, after that I returned node to work and I see that promtail is unable to send data to loki:

```

level=error ts=2018-12-19T08:38:45.168705358Z caller=client.go:110 msg="error sending batch" error="Error doing write: 500 - 500 Internal Server Error"
```

and there are many of such messages

logs of loki:
```
level=warn ts=2018-12-19T08:38:12.877018059Z caller=gokit.go:46 method=/logproto.Pusher/Push duration=48.9µs request="&PushRequest{Streams:[&Stream{Labels:{__filename__=<I cut my real logs here, there was almost 100k of symbols>&Stream{Labels:{__filename__=\"/var/log/pods/393c4005-02b8-11e9-9a39-000d3a393b6c/promtail/0.log\", instance=\"promtail-9vfhm\", job=\"monitoring/promtail\", namespace=\"monitoring\", stream=\"stderr\"},Entries:[{2018-12-19 08:38:12.869851155 +0000 UTC level=error ts=2018-12-19T08:38:12.869702255Z caller=client.go:110 msg=\"error sending batch\" error=\"Error doing write: 500 - 500 Internal Server Error\"\n}],}],}" err="Entry out of order" msg="gRPC\n"0
```

as you can see, loki returned `Entry out of order` error, as far as I understand, the error is caused by this condition:
https://github.com/grafana/loki/blob/master/pkg/chunkenc/gzip.go#L86

which means that loki is unable to add block into memory, does it mean that I need restart of loki after restart of any of promtail instances?

**Comment by gouthamve** [comment]
I think promtail is not picking up from it left off but rather from the beginning again. This should be fixed and further, we should not be returning a 500 rather but a 400 on OOO inserts. See: https://github.com/grafana/loki/issues/168

**Comment by tomwilkie** [comment]
Promtail should pickup where left off, I suspect we're not persisting the posititions.yml to the host FS in the k8s config.  @kovetskiy what config did you use to deploy it?

**Comment by kovetskiy** [comment]
@tomwilkie oh, I was the earliest adopter and I created my own helm chart since there was no one.

And yes, there is no persistent volumes in my promtail daemonset. As for config, I used default one for kubernetes. 

There is only persistent volume only for loki (/var/loki).

**Comment by andrusstrockiy** [comment]
- Got  the same error on all my nodes from Daemon Sets loki-grafana-promtail
- Installation of loki was made using helm chart from https://github.com/grafana/loki/tree/master/production/helm
> suspect we're not persisting the posititions.yml

Actually, I don't see any presistances mentions in Helm chart

**Comment by andridzi** [comment]
got the same issue
after running promtail at first time on a new node, loki restarts and after restarting many pod logs missed
if I restart promtail I got tons of errors ```level=error ts=2019-01-29T21:25:32.497598564Z caller=client.go:110 msg="error sending batch" error="Error doing write: 500 - 500 Internal Server Error"
level=error ts=2019-01-29T21:25:32.504522158Z caller=client.go:110 msg="error sending batch" error="Error doing write: 500 - 500 Internal Server Error"
level=error ts=2019-01-29T21:25:32.514034441Z caller=client.go:110 msg="error sending batch" error="Error doing write: 500 - 500 Internal Server Error"```

**Comment by jmvizcainoio** [comment]
i have similiar problem

```
msg="error sending batch" error="Error doing write: 400 - 400 Bad Request"

**Comment by daixiang0** [comment]
still exist.

**Comment by daixiang0** [comment]
still exist.

**Comment by gangstead** [comment]
Just ran into this problem.  Not sure how to recover.

**Comment by gangstead** [comment]
OK I fixed it by wiping the data directory of the loki pod.  In the stateful set configuration I could see that a volume called `storage` of type `emptyDir` was being mounted to `/data`.  `emptyDirs` persist across restarts as long as the pod is on the same node, so restarting the loki pod is not enough.

I shelled into the pod and removed `/data/loki` then killed it and let it restart.  Now all the promtails are able to send logs over again.  Hope this helps someone with the same problem who wants to get back up and running as simply as possible.
